### PR TITLE
Carousel Finite Upgrades

### PR DESCRIPTION
**Pull request by ClayCooperLA**
Adds `snapbackThreshold`, `maxSnapOvershootVelocity`, `maxWheelVelocity`, to help reduce extreme overshoot in finite carousels.

Additionally an experimental solution for handling mouse wheel inertia when overshooting.
This is called with `engageWheelOvershootTimeout()`, a function that:
Attempts to prevent user from seeing extreme overshoot from a high velocity. User can still move with wheel just not engage inertia during this timeout period. If there is enough inertia on the overshoot users will feel the pullback of the still decaying inertia when the timer expires.

Increasing timeout means more time that the user cannot engage inertia (bad), but less chance of decaying inertia animating the track in the wrong way (good)

As of this version, 750ms timeout feels like a good compromise. Things still get weird if the user purposefully tries to yank at the edges over and over. The alternative is allowing it to fly off the screen before coming back.

Hopefully one day we’ll be able to cancel wheel inertia.

@claus Please let me know how I can improve these additions to better fit with the existing code.

UPDATE 08/2022: Adds `CarouselWrapper` example, comes with `DotNav` and `ArrowNav` components. Also a `maxWidth` mode that puts the carousel in a centered max width section but only offsets the snap position so the carousel remains full bleed but anchored to the grid.

UPDATE 08/30/2022: Adds infinite support for `CarouselWrapper` (not perfect), as well as `overflowFade` and `auto timer` options.

### CODE REVIEW

File: components/pages/demos/CarouselWrapperDemo/CarouselWrapperDemo.js [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+    const [maxWidth, setMaxWidth] = useState(true);
+    const [visibleItems, setVisibleItems] = useState(1);
+
+    const [infinite, setInfinite] = useState(false);
+    const [randomWidths, setRandomWidths] = useState(false);
+    const [overflowFade, setOverflowFade] = useState(false);
+    const [autoTimerSeconds, setAutoTimerSeconds] = useState(0);
+
+    const isLarge = useMedia(`(min-width: 1280px)`, true);
+    const isXLarge = useMedia(`(min-width: 1920px)`, true);
```
Comment (claus) [inline components/pages/demos/CarouselWrapperDemo/CarouselWrapperDemo.js]: nope

File: components/pages/demos/CarouselWrapperDemo/CarouselWrapperDemo.js [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+                        <CarouselWrapper
+                            ref={carouselRef}
+                            snap
+                            align={align}
+                            infinite={infinite}
+                            activeItemIndex={activeItemIndex}
+                            className={styles.carouselKittens}
+                            visibleItems={visibleItems}
+                            snapbackThreshold={isLarge ? 100 : 50}
+                            maxSnapOvershootVelocity={isLarge ? 3 : 2}
```
Comment (claus) [inline components/pages/demos/CarouselWrapperDemo/CarouselWrapperDemo.js]: If `snapbackThreshold` and `maxSnapOvershootVelocity` need to be responsive, they should be CSS vars

File: components/pages/demos/CarouselWrapperDemo/CarouselWrapperDemo.js (hunk 1 above)
Comment (ClayCooperLA) [reply components/pages/demos/CarouselWrapperDemo/CarouselWrapperDemo.js]: 😂

File: components/pages/demos/CarouselWrapperDemo/CarouselWrapperDemo.js (hunk 2 above)
Comment (ClayCooperLA) [reply components/pages/demos/CarouselWrapperDemo/CarouselWrapperDemo.js]: it has been done, so much better!

**Comment by stevescavo** [comment]
@claus @ClayCooperLA Since so many changes have been made to this component and since it's been moved to https://github.com/MadeInHaus/haus-core, I recommend when close this PR.

**Comment by claus** [comment]
I'd like to keep it open for a bit, it has some stuff that i'd like to steal

**Comment by stevescavo** [comment]
> I'd like to keep it open for a bit, it has some stuff that i'd like to steal

ok

**Pull request closed without merging**